components/script/dom/htmlformelement.rs

```diff
@@ -1236,7 +1293,8 @@ pub enum FormMethod {
 pub enum FormSubmitter<'a> {
     FormElement(&'a HTMLFormElement),
     InputElement(&'a HTMLInputElement),
-    ButtonElement(&'a HTMLButtonElement), // TODO: image submit, etc etc
+    ButtonElement(&'a HTMLButtonElement),
+    // TODO: the spec doesn't seem to put hard constraints on submitter, so it can theoretically be any element
```

Comment:
I think instead of here, we can add a doc to the enum with `\\\ <https://html.spec.whatwg.org/multipage/#form-associated-element>`, and then add a TODO that would simply state to implement the other types of form associated elements, including custom elements.

Reply:
ok, done